testlib/s3mock.py

```diff
@@ -43,6 +38,17 @@ class ThouShaltNotPass(Exception):
 }
 
 
+@dataclasses.dataclass
+class Request:
+    method: str
+    url: str
+    body: io.BytesIO
+    headers: dict
+    scheme: str
+    host: str
+    port: int
```

Comment:
This is my first experience with data classes. I like them. It's super easy to do this and you get to skip a ton of boilerplate. Data classes are new in 3.7. We could also use `attrs` which is in the requirements already. That's <3.7 friendly. With antenna, it's not a big deal. With socorro, we have some code we "export" to other libraries that support a wider range of Python versions, so we might want to be more careful.